solutions/fully-configurable/variables.tf

```diff
@@ -203,81 +220,59 @@ variable "cos_bucket_name" {
-  type        = string
-  default     = null
+variable "cos_bucket_access_tags" {
+  type        = list(string)
+  description = "A list of access tags to apply to the Cloud Object Storage bucket created by the module. For more information, see https://cloud.ibm.com/docs/account?topic=account-access-tags-tutorial."
+  default     = []
 
   validation {
-    condition     = var.cross_region_location == null || can(regex("us|eu|ap", var.cross_region_location))
-    error_message = "The variable `cross_region_location` value must be `us` or `eu`, `ap`, or `null`."
+    condition = alltrue([
+      for tag in var.cos_bucket_access_tags : can(regex("[\\w\\-_\\.]+:[\\w\\-_\\.]+", tag)) && length(tag) <= 128
+    ])
+    error_message = "Tags must match the regular expression \"[\\w\\-_\\.]+:[\\w\\-_\\.]+\", see https://cloud.ibm.com/docs/account?topic=account-tag&interface=ui#limits for more details"
   }
 }
 
-variable "cos_bucket_region" {
-  type        = string
-  description = "The COS bucket region. If you pass a value for this variable, you must set the value of `cross_region_location` to null. If `cross_region_location` and `cos_bucket_region` are both set to null, then `region` will be used."
-  default     = null
-}
-
-variable "archive_days" {
-  description = "Specifies the number of days when the archive rule action takes effect. This must be set to null when when using var.cross_region_location as archive data is not supported with this feature."
-  type        = number
-  default     = null
+variable "skip_event_notifications_cos_auth_policy" {
+  type        = bool
+  description = "Set to `true` to skip the creation of an IAM authorization policy that permits the Event Notifications instance `Object Writer` and `Reader` access to the given Object Storage bucket. Set to `true` to use an existing policy."
+  default     = false
 }
 
-variable "retention_enabled" {
+variable "skip_cos_kms_auth_policy" {
   type        = bool
-  description = "Set to `true` to skip the creation of an IAM authorization policy that permits all Event Notifications instances in the resource group to read the encryption key from the KMS instance."
+  description = "Set to true to skip the creation of an IAM authorization policy that permits the COS instance to read the encryption key from the KMS instance. If set to false, pass in a value for the KMS instance in the `existing_key_management_service_instance_crn` variable. If a value is specified for `ibmcloud_kms_api_key`, the policy is created in the KMS account."
   default     = false
 }
 
+variable "cos_bucket_region" {
+  type        = string
+  description = "The COS bucket region. If `cos_bucket_region` is set to null, then `region` will be used."
+  default     = "us-south"
```

Comment:
Can we take same approach as SCC DA where default value will be `null` (meaning it will be provisioned in same region as the EN instance). See https://github.com/terraform-ibm-modules/terraform-ibm-scc/blob/6fbd4973c72855920c157848ab9eee0b7cce5439/solutions/fully-configurable/variables.tf#L301-L305